### PR TITLE
Implement Controller

### PR DESCRIPTION
**Pull request by german-e-mas**
In this PR I implement the controller side of the MVC Pattern.

Updates include:
- Controller class.
- Speed of the Aircraft was changed.
- Radius updates.
- Minor language correction.
- Touch Event to select Aircraft and move them around.

ptal @pablisho

### CODE REVIEW

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 1]
```diff
@@ -24,7 +24,9 @@
  */
 public class Game implements AircraftGenerator.OnAircraftGenerated {
     private static final String TAG = Game.class.getSimpleName();
-    private static final int UPDATE_MS = 200;
+    private static final int UPDATE_MS = 30;
+    // Modifier of the Aircraft radius, in order to give a larger margin of selection.
+    private static final float SIZE_MULTIPLIER = 1.25f;
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: I wouldn't think it this way. I would think it as a tolerance or distance from the centre, similar as the landing.

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 2]
```diff
@@ -86,6 +88,19 @@
...
+         * at a time.
+         *
+         * @param id    ID of the selected Aircraft.
+         */
+        void onAircraftSelect(int id);
+
+        /**
+         * A specific aircraft has been deselected.
+         */
+        void onAircraftDeselect(int id);
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: Maybe better to have only one callback and pass a boolean "selected" as param indicating if it was selected or deselected.

File: app/src/main/java/mas/german/landingplanes/Game.java [hunk 3]
```diff
@@ -261,6 +286,65 @@ public Aerodrome getAerodrome() {
...
+    /**
+     * Auxiliary method to select a single aircraft while deselecting the rest.
+     *
+     * @param id    ID of the Aircraft to select.
+     */
+    private void selectAircraft(int id) {
+        synchronized (mAircraftList) {
+            for (Aircraft aircraft : mAircraftList) {
+                if (aircraft.getId() == id) {
+                    aircraft.select();
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/Game.java]: same here, maybe aircraft.select(true) aircraft.select(false) could be better

File: app/src/main/java/mas/german/landingplanes/MainActivity.java [hunk 4]
```diff
@@ -2,15 +2,21 @@
...
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import mas.german.landingplanes.controller.Controller;
+import mas.german.landingplanes.view.AerodromeView;
 
 public class MainActivity extends AppCompatActivity {
     private static final String TAG = MainActivity.class.getSimpleName();
+    private AerodromeView mView;
     private Game mGame = Game.getInstance();
+    private Controller mCtrl = new Controller();
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/MainActivity.java]: mController

File: app/src/main/java/mas/german/landingplanes/MainActivity.java [hunk 5]
```diff
@@ -2,15 +2,21 @@
...
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+        mView = (AerodromeView) findViewById(R.id.view_aerodrome);
+        mView.setListener(mCtrl);
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/MainActivity.java]: setController

File: app/src/main/java/mas/german/landingplanes/view/AerodromeView.java [hunk 6]
```diff
@@ -131,6 +167,30 @@ public void onAircraftPositionChanged() {
...
 
   @Override
+  public void onAircraftSelect(int id) {
+    synchronized (mDrawables.values()) {
+      // Select the aircraft with the matched ID, while deselecting the rest.
+      for (AircraftDrawable aircraftDrawable : mDrawables.values()) {
+        if (aircraftDrawable.getId() == id) {
+          aircraftDrawable.select();
+        } else {
+          aircraftDrawable.deselect();
```
Comment (pablisho) [inline app/src/main/java/mas/german/landingplanes/view/AerodromeView.java]: same about select/deselect

File: app/src/main/java/mas/german/landingplanes/Game.java (hunk 1 above)
Comment (german-e-mas) [inline app/src/main/java/mas/german/landingplanes/Game.java]: I agree, it is more of a tolerance than a size modifier. `DISTANCE_TOLERANCE` seems like a better name for what it does.

**Comment by german-e-mas** [comment]
I addressed the comments @pablisho . I unified the selection and de-selection methods of both `Aircraft` and `AircraftDrawable`, along various comments and renames.

ptal again.

**Comment by german-e-mas** [comment]
I changed the distance tolerance to an addition, not a multiplication.

ptal @pablisho

**Pull request merged**